internal/pkg/machine/events/events.go

```diff
@@ -65,28 +68,33 @@ func NewHandler(ctx context.Context, st state.State, machineIndex int) (*Handler
 			var dialer net.Dialer
 
 			if bindAddress == nil {
-				var addr *network.NodeAddress
+				var list safe.List[*network.AddressStatus]
 
-				addr, err = safe.ReaderGetByID[*network.NodeAddress](ctx, st, network.NodeAddressDefaultID)
+				list, err = safe.ReaderListAll[*network.AddressStatus](ctx, st)
 				if err != nil {
 					return nil, err
 				}
 
-				if len(addr.TypedSpec().Addresses) == 0 {
+				addr, ok := list.Find(func(r *network.AddressStatus) bool {
+					return strings.HasPrefix(r.TypedSpec().LinkName, constants.SideroLinkName)
+				})
+				if !ok {
 					return nil, fmt.Errorf("failed to look up siderolink address")
 				}
 
-				siderolinkAddr := addr.TypedSpec().Addresses[0]
+				siderolinkAddr := addr.TypedSpec().Address
 
 				bindAddress = net.TCPAddrFromAddrPort(netip.AddrPortFrom(
 					siderolinkAddr.Addr(),
 					0,
 				))
+
+				linkName = addr.TypedSpec().LinkName
 			}
 
 			dialer.LocalAddr = bindAddress
 
-			dialer.Control = emunet.BindToInterface(fmt.Sprintf("%s%d", constants.SideroLinkName, machineIndex))
+			dialer.Control = emunet.BindToInterface(linkName)
```

Comment:
I think its a good idea to move `linkName` in this function instead of outer scope.